src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -4299,6 +4299,25 @@ VOID    MethodTableBuilder::InitializeFieldDescs(FieldDesc *pFieldDescList,
                 pwalk++;
             }
         }
+        else
+        {
+            ULONG fieldOffset = 0;
+            if (pInternalImport->GetFieldOffset(bmtMetaData->pFields[i], &fieldOffset) == S_OK)
```

Comment:
This is an extra binary search in metadata tables for every field. Can we do the binary search just once per type?

I am wondering whether we need to check this at all. I think silently ignoring the explicit offsets for auto-layout classes like what are doing today is fine. We do not generally check for silently ignored attributes.

Reply:
I'm fine removing this from auto-layout and changing the breaking change to sequential only.

I wanted to start with a wider net and narrow it down if requested.

Reply:
I think we can do the search once per type, just not with the APIs on `IMDInternalImport`. I'll see if I can do that cleanly.

Reply:
What do you think about not doing this at all and just silently ignoring the explicit offsets for types without explicit layout?

Reply:
It looks like writing a more efficient implementation here is remarkably difficult with our current metadata API. We'd have to write a new implementation and we need a primitive of "binary search for the closest entry", which the current implementation doesn't have.

We can't even use `GetClassLayoutInit` and `GetClassLayoutNext` as that does the binary search each time.

I'd happily change this to just silently ignore the offsets for non-explicit types completely. @davidwrighton suggested adding the error for sequential types as we're removing a (compeletely unused) feature. I wanted to expand it to validate for all types, but I think that's impractical at this point.

Let me know if you want this to be done for sequential-only or not at all.

Reply:
My preference is not at all.

I am really only worried about cases that can be produced by C# or other mainstream high-level compiler. I am not worried about hand-written IL.

Reply:
Sounds great to me!

Reply:
I've removed the error checking for sequential layout as well. The new behavior is to ignore offsets for all non-explicit layouts as they can only be done so with hand-written IL.